spec/controllers/assignments_controller_spec.rb

```diff
@@ -59,26 +63,27 @@
     end
     it 'redirects to the index' do
       submit
-      expect(response).to redirect_to assignments_url
+      expect(response).to redirect_to roster_assignments_url
     end
   end
 
   describe 'GET #edit' do
     before :each do
       @assignment = create :assignment
+      @roster = create :roster
       when_current_user_is :whoever
     end
     let :submit do
-      get :edit, id: @assignment.id
+      get :edit, roster_id: @roster.id, id: @assignment.id
     end
     it 'finds the correct assignment' do
       submit
       expect(assigns.fetch :assignment).to eql @assignment
     end
     it 'populates a users variable of all users' do
-      user_1 = create :user
-      user_2 = create :user
-      user_3 = create :user
+      user_1 = create :user, rosters: [@roster] 
```

Comment:
I don't understand your answer.

My question was whether it would be worth writing a method e.g.:

```ruby
def roster_user(roster)
  create :user, rosters: [roster]
end
```

So that we could cut down on the repetition here, by calling e.g.:

```
user_1 = roster_user(@roster)
```